.github/workflows/build-yocto.yml

```diff
@@ -115,3 +115,35 @@ jobs:
           (( retries == i )) && { echo 'Failed to publish artifacts'; exit 1; }
 
           echo Image available at: ${url}
+  boot-test:
```

Comment:
Your commit says that you are adding to nightly builds, but you are effectively adding it to each action (merge to main, PR, and nightly).

Reply:
I don't know if it should run on all builds. I can change either way.

Reply:
We should probably run on all builds for now, if it ends up being too much we can split things later.

Reply:
lol. ricardo replied 1 minute faster than me ;)

Reply:
yes, and we have a draw 1:1 :)

Reply:
@mwasilew you decide :D

Reply:
I think I have to limit the boot test to only merges and nightly builds. By default PRs don't have access to org/repository secrets.

Reply:
I added new condition. When build is triggered by PR, the boot test step is skipped.